hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -210,16 +210,19 @@ public RpcClient(ConfigurationSource conf, String omServiceId)
       ozoneManagerProtocolClientSideTranslatorPB.setS3AuthCheck(isS3);
       if (isS3) {
         // S3 Auth works differently and needs OM version to be at 2.0.0
-        String omVersion = conf.get(OZONE_OM_CLIENT_PROTOCOL_VERSION_KEY);
-        if (!validateOmVersion(omVersion, serviceInfoEx.getServiceInfoList())) {
+        OzoneManagerVersion minOmVersion = conf.getEnum(
+            OZONE_CLIENT_REQUIRED_OM_VERSION_MIN_KEY,
+            OzoneManagerVersion.DEFAULT_VERSION);
+        if (!validateOmVersion(
+            minOmVersion, serviceInfoEx.getServiceInfoList())) {
           if (LOG.isDebugEnabled()) {
             for (ServiceInfo s : serviceInfoEx.getServiceInfoList()) {
               LOG.debug("Node {} version {}", s.getHostname(),
-                  s.getProtobuf().getOMProtocolVersion());
+                  s.getProtobuf().getOMVersion());
             }
           }
           throw new RuntimeException("OmVersion expected "
-              + omVersion
+              + minOmVersion
```

Comment:
Let's update the message to indicate the minimum version requirement was not met. Right now it reads like an exact match is required.